uni/lib/model/entities/session.dart

```diff
@@ -19,21 +20,25 @@ class Session {
 
   /// Creates a new Session instance from an HTTP response.
   /// Returns null if the authentication failed.
-  static Session? fromLogin(
+  static Future<Session?> fromLogin(
     http.Response response,
     List<String> faculties, {
     required bool persistentSession,
-  }) {
+  }) async {
     final responseBody = json.decode(response.body) as Map<String, dynamic>;
 
     if (!(responseBody['authenticated'] as bool)) {
       return null;
     }
 
-    return Session(
+    final session = Session(
       faculties: faculties,
       username: responseBody['codigo'] as String,
       cookies: NetworkRouter.extractCookies(response.headers),
     );
+
+    session.faculties = await getStudentFaculties(session);
```

Comment:
Alright, so the drop-down choice is overriden. That UI path should be completely removed.